### ISSUE TITLE
About creating a Duration instance

### ISSUE THREAD

**Issue by Chiarorucs** [snowdrop/istio-java-api#113]
Hey! I'm the #110 gay.I'm afraid I'll bother you again. About creating a Duration instance.

There're some fields with type "Duration" among Destinationrule settings, such as "interval"

`interval   |  Duration   |  Time interval between ejection sweep analysis. format: 1h/1m/1s/1ms. MUST BE >=1ms. Default is 10s.`

The Duration.class has a constructor like 
`public Duration(Integer nanos, Long seconds) {
        this.nanos = nanos;
        this.seconds = seconds;
    }`
If I want to set the field interval as "1h/1m/1s/1ms", what should I do with these two params : nanos and seconds ?

Also,I'm aware of the inner class Deserializer and its method deserialize(). 

`public Duration deserialize(JsonParser parser, DeserializationContext ctxt) throws IOException, JsonProcessingException {`

         `   ObjectCodec oc = parser.getCodec();
            JsonNode node = (JsonNode)oc.readTree(parser);
            Period period = Duration.FORMATTER.parsePeriod(node.asText());
            return new Duration(0, (long)period.toStandardSeconds().getSeconds());
        }`

In it, I've paid attention to the field FORMATTER which seems like turning time and its unit to seconds.

`private static final PeriodFormatter FORMATTER = (new PeriodFormatterBuilder()).printZeroNever().appendHours().appendSuffix("h").appendMinutes().appendSuffix("m").appendSeconds().appendSuffix("s").appendMillis().appendSuffix("ms").toFormatter();`  

Should I use the method deserialize()? How to create and initialize these two params :parser and ctxt?

Or how to make it easier?

Thank you so much!

**Comment by shalk** [comment]
I meet the same issue  when I use use  `timeout` and `fault inject`( delay) for virtualservice。
When i set timeout as  1200 ms =  1s   200_000_000 nanos，  the nanos will be ignore， only 1s applied 
I seens  the  Serializer  or  Deserializer only support `seconds`
```
//me.snowdrop.istio.api.Duration.java

    public static class Serializer extends JsonSerializer<Duration> {
        @Override
        public void serialize(Duration value, JsonGenerator gen, SerializerProvider serializers) throws IOException {
            gen.writeString(FORMATTER.print(Seconds.seconds(value.seconds.intValue())));
        }
    }
```

**Comment by metacosm** [comment]
I will try to address this issue this week. Sorry for the inconvenience.

**Comment by shalk** [comment]
> I will try to address this issue this week. Sorry for the inconvenience.

Thank you for reply. In my case , millisecond is accurate enough.  my istio-java-api's  version is  1.6.5-beta4 while our istio is 1.5.x

**Comment by metacosm** [comment]
This should be fixed with the release of 1.6.14.
